lib/modules/manager/helm-values/extract.ts

```diff
@@ -9,20 +9,31 @@ import {
   matchesHelmValuesInlineImage,
 } from './util';
 
-function getHelmDep({
-  registry,
-  repository,
-  tag,
-}: {
-  registry: string;
-  repository: string;
-  tag: string;
-}): PackageDependency {
+let tagIsNumber = false;
+
+function getHelmDep(
+  {
+    registry,
+    repository,
+    tag,
+  }: {
+    registry: string;
+    repository: string;
+    tag: string;
+  },
+  isNumber: boolean
+): PackageDependency {
   const dep = getDep(`${registry}${repository}:${tag}`, false);
   dep.replaceString = tag;
   dep.versioning = dockerVersioning;
-  dep.autoReplaceStringTemplate =
-    '{{newValue}}{{#if newDigest}}@{{newDigest}}{{/if}}';
+  if (isNumber) {
+    dep.autoReplaceStringTemplate =
+      '"{{newValue}}{{#if newDigest}}@{{newDigest}}{{/if}}"';
+    tagIsNumber = false;
+  } else {
+    dep.autoReplaceStringTemplate =
+      '{{newValue}}{{#if newDigest}}@{{newDigest}}{{/if}}';
+  }
```

Comment:
As told before, we should always quote because the new tag can be a number so we will write a wrong values later with autoreplace

Reply:
@viceice as I saw if the tag is already wrapped with double quotes the new version also will be wrapped with double quotes, 
and if it is wrapped with a single quote the new version will be wrapped with single quotes(this behavior already exists). 
so I just supported the case when we have a tag without any quotes I added double quotes to the new version.